src/analyser.cpp

```diff
@@ -0,0 +1,1509 @@
+/*
+Copyright libCellML Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "libcellml/analyser.h"
+
+#include "libcellml/analyserequation.h"
+#include "libcellml/analyserequationast.h"
+#include "libcellml/analysermodel.h"
+#include "libcellml/analyservariable.h"
+#include "libcellml/component.h"
+#include "libcellml/model.h"
+#include "libcellml/units.h"
+#include "libcellml/validator.h"
+#include "libcellml/variable.h"
+
+#include "analyserequation_p.h"
+#include "analyserequationast_p.h"
+#include "analysermodel_p.h"
+#include "analyservariable_p.h"
+#include "utilities.h"
+#include "xmldoc.h"
+
+#ifdef __linux__
+#    undef TRUE
+#    undef FALSE
+#endif
+
+namespace libcellml {
+
+static const size_t MAX_SIZE_T = std::numeric_limits<size_t>::max();
+
+struct AnalyserInternalEquation;
+using AnalyserInternalEquationPtr = std::shared_ptr<AnalyserInternalEquation>;
+using AnalyserInternalEquationWeakPtr = std::weak_ptr<AnalyserInternalEquation>;
+
+struct AnalyserInternalVariable
+{
+    enum struct Type
+    {
+        UNKNOWN,
```

Comment:
Elsewhere these have been alphabetical? Or is that not a thing?

Reply:
I am not sure whether it's a thing, but it was certainly already like that in the old `Generator` class. Also, I like to order enums in a way that makes sense to me (subjective though!).